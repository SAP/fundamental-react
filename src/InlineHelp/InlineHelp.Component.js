import { InlineHelp } from '../';
import React from 'react';
import { Description, DocsText, DocsTile, Header, Import, Properties, Separator } from '../_playground';


export const InlineHelpComponent = () => {
    const defaultHelpPlacement = '<InlineHelp text="Lorem ipsum dolor sit amet, consectetur adipiscing." placement="bottom-right"/>';
    const bottomLeftHelpPlacement = '<InlineHelp text="Lorem ipsum dolor sit amet, consectetur adipiscing." placement="bottom-left"/>';
    const rightHelpPlacement = '<InlineHelp text="Lorem ipsum dolor sit amet, consectetur adipiscing." placement="right"/>';
    const leftHelpPlacement = '<InlineHelp text="Lorem ipsum dolor sit amet, consectetur adipiscing." placement="left"/>';
    const centerHelpPlacement = '<InlineHelp text="Lorem ipsum dolor sit amet, consectetur adipiscing." placement="bottom-center" />';

    return (
        <div>
            <Header>Inline Help</Header>
            <Description>
                Inline help is used to display help text in a popover, often inline with headers, body text and form
                labels.
            </Description>
            <Import module='InlineHelp' path='/fundamental-react/src/' />

            <Separator />

            <Properties
                properties={[
                    {
                        name: 'text',
                        description: 'string (required) - The text to display in the inline help pop-up.'
                    },
                    {
                        name: 'placement',
                        description: 'string (required) - Location for where to display the inline help pop-up. Options include:  \'bottom-right\', \'bottom-left\', \'bottom-center\', \'right\', and \'left\' '
                    }
                ]}
                type='Inputs' />

            <Separator />

            <h2>Default Position</h2>
            <Description>The default positioning of inline help component is bottom right.</Description>
            <DocsTile centered>
                Bottom Right (Default) &nbsp;
                <InlineHelp placement='bottom-right' text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
            </DocsTile>
            <DocsText>{defaultHelpPlacement}</DocsText>

            <Separator />

            <h2>Bottom Left Position</h2>
            <DocsTile centered>
                Bottom Left &nbsp;
                <InlineHelp placement='bottom-left' text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
            </DocsTile>
            <DocsText>{bottomLeftHelpPlacement}</DocsText>

            <Separator />

            <h2>Bottom Center Position</h2>
            <DocsTile centered>
                Bottom Center &nbsp;
                <InlineHelp placement='bottom-center' text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
            </DocsTile>
            <DocsText>{centerHelpPlacement}</DocsText>

            <Separator />

            <h2>Right Position</h2>
            <DocsTile centered>
                Right &nbsp;
                <InlineHelp placement='right' text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
            </DocsTile>
            <DocsText>{rightHelpPlacement}</DocsText>

            <Separator />

            <h2>Left Position</h2>
            <DocsTile centered>
                Left &nbsp;
                <InlineHelp placement='left' text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
            </DocsTile>
            <DocsText>{leftHelpPlacement}</DocsText>

            <Separator />
        </div>
    );
};
