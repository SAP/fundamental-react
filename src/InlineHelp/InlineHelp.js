import PropTypes from 'prop-types';
import React from 'react';

export const InlineHelp = ({ text, placement, className, ...props }) => {
    return (
        <span className='fd-inline-help'>
            <span
                className={`fd-inline-help__content fd-inline-help__content--${placement}${className ? ' ' + className : ''}`} {...props}>
                {text}
            </span>
        </span>
    );
};


InlineHelp.propTypes = {
    placement: PropTypes.oneOf([
        'bottom-right', 'bottom-left', 'right', 'left', 'bottom-center'
    ]).isRequired,
    text: PropTypes.string.isRequired,
    className: PropTypes.string
};
