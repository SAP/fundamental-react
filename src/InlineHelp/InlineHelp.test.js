import { InlineHelp } from './InlineHelp';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<InlineHelp />', () => {
    const defaultInlineHelp = (
        <InlineHelp
            className='blue'
            placement='bottom-right'
            text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
    );

    const inlineHelpBottomLeft = (
        <InlineHelp
            placement='bottom-left'
            text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
    );

    const inlineHelpBottomCenter = (
        <InlineHelp
            placement='bottom-center'
            text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
    );

    const inlineHelpLeft = (
        <InlineHelp
            placement='left'
            text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
    );

    const inlineHelpRight = (
        <InlineHelp
            placement='right'
            text='Lorem ipsum dolor sit amet, consectetur adipiscing.' />
    );

    test('create default InlineHelp component', () => {
        const component = renderer.create(defaultInlineHelp);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('create Bottom Left InlineHelp component', () => {
        const component = renderer.create(inlineHelpBottomLeft);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('create Bottom Center InlineHelp component', () => {
        const component = renderer.create(inlineHelpBottomCenter);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('create Left InlineHelp component', () => {
        const component = renderer.create(inlineHelpLeft);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('create Right InlineHelp component', () => {
        const component = renderer.create(inlineHelpRight);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the InlineHelp component', () => {
            const element = mount(
                <InlineHelp
                    data-sample='Sample'
                    placement='bottom-center'
                    text='Text' />
            );

            expect(
                element.getDOMNode().querySelector('span').attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
