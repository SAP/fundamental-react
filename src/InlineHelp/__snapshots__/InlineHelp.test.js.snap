// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineHelp /> create Bottom Center InlineHelp component 1`] = `
<span
  className="fd-inline-help"
>
  <span
    className="fd-inline-help__content fd-inline-help__content--bottom-center"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing.
  </span>
</span>
`;

exports[`<InlineHelp /> create Bottom Left InlineHelp component 1`] = `
<span
  className="fd-inline-help"
>
  <span
    className="fd-inline-help__content fd-inline-help__content--bottom-left"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing.
  </span>
</span>
`;

exports[`<InlineHelp /> create Left InlineHelp component 1`] = `
<span
  className="fd-inline-help"
>
  <span
    className="fd-inline-help__content fd-inline-help__content--left"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing.
  </span>
</span>
`;

exports[`<InlineHelp /> create Right InlineHelp component 1`] = `
<span
  className="fd-inline-help"
>
  <span
    className="fd-inline-help__content fd-inline-help__content--right"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing.
  </span>
</span>
`;

exports[`<InlineHelp /> create default InlineHelp component 1`] = `
<span
  className="fd-inline-help"
>
  <span
    className="fd-inline-help__content fd-inline-help__content--bottom-right blue"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing.
  </span>
</span>
`;
